File: 785/cpp/main.cpp
#include<iostream>
#include<cstdio>
using namespace std;

const int N = 1e6+10;
int a[N];
int n = 0;

void quickSort(int a[], int l, int r) {
    if( l >= r) {
        return;
    }
    int x = a[(l+r)/2], i = l-1, j = r+1;
    while( i < j) {
        do i++; while(a[i] < x);
        do j--; while(a[j] > x);
        if ( i < j) {
            swap(a[i], a[j]);
        }
    }
    quickSort(a, l, j);
    quickSort(a, j+1, r);
}

int main() {
    scanf("%d", &n);
    for ( int i = 0 ; i < n ; i ++ ) {
        scanf("%d", &a[i]);
    }

    quickSort(a, 0, n-1);
    for ( int i = 0 ; i < n ; i ++ ) {
        printf("%d ", a[i]);
    }
}

File: 785/go/main.go
package main

import "fmt"

func QuickSort(a []int, l, r int) {
	if l >= r {
		return
	}
	x, i, j := a[(l+r)>>1], l-1, r+1
	for i < j {
		i++
		for a[i] < x {
			i++
		}
		j--
		for a[j] > x {
			j--
		}
		if i < j {
			a[i], a[j] = a[j], a[i]
		}
	}
	QuickSort(a, l, j)
	QuickSort(a, j+1, r)
}

func main() {
	var n int
	fmt.Scanf("%d", &n)
	a := make([]int, 0)
	for i := 0; i < n; i++ {
		x := 0
		fmt.Scan(&x)
		a = append(a, x)
	}
	QuickSort(a, 0, n-1)
	for i := 0; i < n; i++ {
		fmt.Printf("%d ", a[i])
	}
}
